gril/grilreply.c

```diff
@@ -873,13 +888,16 @@ GSList *g_ril_reply_parse_get_calls(GRil *gril, const struct ril_msg *message)
 
 		ofono_call_init(call);
 		call->status = parcel_r_int32(&rilp);
-		call->id = parcel_r_int32(&rilp);
+		call->id = parcel_r_int32(&rilp) & 0xFF;
 		call->phone_number.type = parcel_r_int32(&rilp);
 		parcel_r_int32(&rilp); /* isMpty */
 		parcel_r_int32(&rilp); /* isMT */
 		parcel_r_int32(&rilp); /* als */
 		call->type = parcel_r_int32(&rilp); /* isVoice */
-		if (g_ril_vendor(gril) == OFONO_RIL_VENDOR_SAMSUNG_MSM_822x) {
+		if (
```

Comment:
Please try to use less lines here, and dont put x ==y in brackets, since == binds always before && and || (see https://en.cppreference.com/w/c/language/operator_precedence)

Reply:
Sure, corrected in 27ed62a5
Do you have coding guidelines? I am more used to "explicit is better" when operator precedence is involved